pyntc/devices/eos_device.py

```diff
@@ -364,25 +365,49 @@ def set_boot_options(self, image_name, **vendor_specifics):
                 message="Setting install source did not yield expected results",
             )
 
-    def show(self, command, raw_text=False):
-        try:
-            response_list = self.show_list([command], raw_text=raw_text)
-        except CommandListError as e:
-            raise CommandError(e.command, e.message)
+    def show(self, commands, raw_text=False):
+        """Send configuration commands to a device.
 
-        return response_list[0]
+        Args:
+            commands (str, list): String with single command, or list with multiple commands.
+            raw_text (bool, optional): False if encode should be json, True if encoding is text. Defaults to False.
 
-    def show_list(self, commands, raw_text=False):
+        Raises:
+            CommandError: Issue with the command provided.
+            CommandListError: Issue with a command in the list provided.
+        """
         if raw_text:
             encoding = "text"
         else:
             encoding = "json"
 
+        if isinstance(commands, str):
+            command_list = dict(type="string", commands=[commands])
+        else:
+            command_list = dict(type="list", commands=commands)
+
         try:
-            return self._parse_response(self.native.enable(commands, encoding=encoding), raw_text=raw_text)
+            response_list = self._parse_response(
+                self.native.enable(command_list.get("commands"), encoding=encoding), raw_text=raw_text
+            )
```

Comment:
```suggestion
            response = self.native.enable(commands, encoding=encoding)
            response_list = self._parse_response(response, raw_text=raw_text )
```
I formatted this per my suggestion above, but regardless of what is accepted there, this should be split into two separate steps